### ISSUE TITLE
The DepthwiseConv2dNative() function ignores the dilations argument

### ISSUE THREAD

**Issue by PavelPr** [tensorflow/tensorflow#19992]
Hi!

I am using Tensorflow v1.7.0.

I am invoking the DepthwiseConv2dNative() function with a dilations argument that is [1, 2, 2, 1]. Despite of this, the dilations value is being ignored.

Looking at the tensorflow source code, it is evident that the dilations argument is presumably being disregarded (depthwise_conv_op.cc, around line 400) in both CUDNN and non-CUDNN scenarios. Yet, there is no mention of this in the documentation.

Thanks!

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by PavelPr** [comment]
Hi!

Apologies for not including the required information to begin with. I am using TensorFlow under Windows 10, in CPU mode. Bazel version is 0.14.1. 

While I am fully aware of the fact that Windows is not an officially supported platform, reading through the TensorFlow source code creates an impression that this is a yet-to-be-implemented feature. Still, the documentation bears no trace of this (to the contrary, the "dilations" argument is even being described in detail; see: https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/depthwise-conv2d-native).

Have a look at line 400 of depthwise_conv_op.cc (under tensorflow/tensorflow/core/kernels/):

      // TODO(yangzihao): Send in arbitrary dilation rates after the dilated
      // conv is supported.
      launcher_(context, use_cudnn_, cudnn_use_autotune_, input,
                reshaped_filter, /*row_dilation=*/1, /*col_dilation=*/1,
                stride_, stride_, padding_, output, data_format_);

While this particular block pertains to CUDNN, there is no trace of dilation information being passed in the CPU variant, the code for which is right below that block.

Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Shenghsin** [comment]
I have found the same problem， with tf.nn.depthwise_conv2d_native（）， the parameter 'dilations' do not take effect，dilations=[1, 2, 2, 1] will have the same result with dilations=[1, 1, 1, 1].  @PavelPr  had pointed the bug in tensorflow's  implementation.

**Comment by karmel** [comment]
@chsigg @yzhwang -- are there current plans to add dilation parameters to DepthwiseCov2dNative? If not, should the docs be updated?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by olesalscheider** [comment]
This definitely sounds like a problem that can reduce the accuracy of your models, but it might go unnoticed by some.

If the cudnn path does not support the dilated convolution in this case yet, maybe it would be a good idea to just use the native path if the dilation rate is > 1?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 90 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Rasterer** [comment]
Meet the same issue. Any update? If this can not be supported in short term, I think at least the doc should be updated first, it took me much time to figure out there is a misalignment between doc and real behavior.

**Comment by jvishnuvardhan** [comment]
@PavelPr  Is this still an issue with the latest TF1.13 and TF2.0? Thanks!

**Comment by jvishnuvardhan** [comment]
Closing due to lack of recent activity, but please let me know if I'm mistaken. Since this issue is old at this point, please reopen the issue if it still occurs when tried with the latest version of Tensorflow. Thank you.

**Comment by weichiche** [comment]
@jvishnuvardhan The issue still exists in v1.13.1.

**Comment by jvishnuvardhan** [comment]
@weichiche Could you create new issue with the issue details and context. Thanks!

**Comment by SuperShinyEyes** [comment]
I just stumbled on this thread while looking at [`tf.nn.depthwise_conv2d`](https://www.tensorflow.org/api_docs/python/tf/nn/depthwise_conv2d). I assume `tf.nn.depthwise_conv2d` is a wrapper for `tensorflow::ops::DepthwiseConv2dNative`, and I will share my experiments in hope it is useful for someone.

I'm on Linux, testing on CPU mode with tensorflow 1.14.0(https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-1.14.0-cp37-cp37m-linux_x86_64.whl), and dilation seems to work. Below is a self-contained script for testing. 
```python
%reload_ext autoreload
%autoreload 2
%matplotlib inline

import matplotlib.pyplot as plt
import numpy as np

import tensorflow as tf
tf.enable_eager_execution()

x = np.random.rand(32,64,3);x.shape

plt.figure()
plt.title('X')
plt.imshow(x)

num_channels = 3
box_filter = np.ones([2, 2, 3, 1]) / 4
dilation_size = 2

y_with_dilation = tf.nn.depthwise_conv2d(
    np.expand_dims(x, axis=0),
    box_filter,  # [filter_height, filter_width, in_channels, out_channels]
    dilations=[dilation_size, dilation_size],
    strides=[1,1,1,1],
    padding="VALID",   # No padding
    data_format="NHWC"
)

y_without_dilation = tf.nn.depthwise_conv2d(
    np.expand_dims(x, axis=0),
    box_filter,  # [filter_height, filter_width, in_channels, out_channels]
    strides=[1,1,1,1],
    padding="VALID",   # No padding
    data_format="NHWC"
)

def plot_y(y, title=''):
    y = tf.squeeze(y)
    plt.figure()
    plt.title(title)
    plt.imshow(y)
    
plot_y(y_with_dilation, 'With dilation')
plot_y(y_without_dilation, 'Without dilation')
```
![x](https://user-images.githubusercontent.com/5738363/60093996-8458c280-9753-11e9-851c-cd3e516f1142.png)
![1](https://user-images.githubusercontent.com/5738363/60093999-8753b300-9753-11e9-9e49-f58661bdac99.png)
![2](https://user-images.githubusercontent.com/5738363/60094005-89b60d00-9753-11e9-8112-4c1d590ccb71.png)

Please don't ask me why I'm using a uniform filter :)
